cesium/featurize.py

```diff
@@ -135,53 +152,115 @@ def featurize_time_series(times, values, errors=None, features_to_use=[],
+            times = [np.tile(t, (len(v), 1)) for t, v in zip(times, values)]
+        assert (compatible_shapes(values, times) and
+                compatible_shapes(values, errors))
+    except:
+        raise ValueError("times, values, errors are not of compatible"
+                         " types/sizes. Please refer to the docstring"
+                         " for list of allowed input types.")
+
     if labels is None:
         if isinstance(times, (list, tuple)):
             labels = np.arange(len(times))
         else:
             labels = np.array([0])
 
-    if all([isinstance(x, np.ndarray) for x in (times, values, errors)]):
-        times, values, errors = ([times], [values], [errors])
     if isinstance(meta_features, pd.Series):
         meta_features = meta_features.to_dict()
     if targets is not None:
         targets = pd.Series(targets, index=labels)
 
-    if not all([isinstance(x, (list, tuple))
-                for x in (times, values, errors)]):
-        raise TypeError("times, values, and errors have incompatible types")
-
     meta_features = pd.DataFrame(meta_features, index=labels)
 
-    if use_celery:
-        if not celery_available():
-            raise RuntimeError("Celery unavailable; please check your Celery "
-                               "configuration or set `use_celery=False`.")
-
-        all_time_series = [TimeSeries(t, m, e, meta_features.loc[label],
-                                      name=label)
-                           for t, m, e, label in zip(times, values, errors,
-                                                     labels)]
-        celery_res = group(featurize_data_task.s(ts, features_to_use,
-                                                 custom_script_path,
-                                                 custom_functions)
-                           for ts in all_time_series)()
-        res_list = celery_res.get()
-        labels, feature_dicts = zip(*res_list)
-        if targets is not None:
-            targets = targets.loc[list(labels)]
-        meta_features = meta_features.loc[list(labels)]
-        meta_feature_dicts = meta_features.to_dict(orient='record')
-    else:
-        feature_dicts = []
-        meta_feature_dicts = []
-        for t, m, e, label in zip(times, values, errors, labels):
-            ts = TimeSeries(t, m, e, meta_features=meta_features.loc[label])
-            feature_dict = ft.featurize_single_ts(ts, features_to_use,
-                                                  custom_script_path=custom_script_path,
-                                                  custom_functions=custom_functions)
-            feature_dicts.append(feature_dict)
-            meta_feature_dicts.append(ts.meta_features)
-    return ft.assemble_featureset(feature_dicts, targets, meta_feature_dicts,
-                                  labels)
+    all_time_series = [delayed(TimeSeries(t, m, e, meta_features.loc[label],
```

Comment:
Since we convert to TImeSeries anyway, why not just convert everything to TimeSeries, and then check those shapes against one another?